packages/web/lib/hooks/useBoundingRect.ts

```diff
@@ -0,0 +1,71 @@
+import { MutableRefObject, useEffect, useRef, useState } from 'react';
+
+const debounce = (
```

Comment:
Do we not have something like this in utils already? If not this should probably go there

Reply:
Not sure. I copied this working code from another repo of mine without looking into it 😅